src/runtime/policy/tests/policy-test.ts

```diff
@@ -874,6 +874,31 @@ policy MyPolicy {
       ingressValidation.getMaxReadType(typeVar), expected);
   });
 
+  it('uses resolved type of a typevar to get max read type', async () => {
+    const manifest = await Manifest.parse(`
+      schema A
+        foo: inline Foo {a: Text, b: Text, c: Text, d: Text}
+
+      policy P {
+        from A access {
+          foo {a, b}
+        }
+      }
+    `);
+    const ingressValidation = new IngressValidation(manifest.policies);
+    const typeVar = await createTypeVarForSchema(
+      'A',
+      'foo: inline Foo {a: Text, d: Text}',
+      'foo: inline Foo {a: Text, d: Text}');
+    typeVar.maybeEnsureResolved();
```

Comment:
do you want to add an assert also before the call to `maybeEnsureResolved()` to verify what the getMaxReadType returns then? or is it a separate test case?

Reply:
There are separate test cases for unresolved type variables.